libbeat/common/mapstr.go

```diff
@@ -380,3 +389,43 @@ func (op putOperation) Do(key string, data MapStr) (interface{}, error) {
 	data[key] = op.Value
 	return existingValue, nil
 }
+
+type setValueOperation struct {
+	Value interface{}
+}
+
+func (op setValueOperation) Do(key string, data MapStr) (interface{}, error) {
+	if op.Value == nil {
+		return data, nil
+	}
+
+	switch op.Value.(type) {
+	case *bool:
+		if newVal := op.Value.(*bool); newVal != nil {
```

Comment:
These secondary type assertions are not necessary if the switch statement is modified to be `switch v := op.Value.(type)` (see [TypeSwitchGuard](https://golang.org/ref/spec#Type_switches)). Within each case statement `v` takes on the type declared in the `case`.